mathesar_ui/src/stores/tables.ts

```diff
@@ -273,22 +292,57 @@ export function getTable(id: TableEntry['id']): CancellablePromise<TableEntry> {
   return getAPI(`/api/db/v0/tables/${id}/`);
 }
 
+export function splitTable(
+  id: number,
+  idsOfColumnsToExtract: number[],
+  extractedTableName: string,
+): CancellablePromise<SplitTableResponse> {
+  return postAPI(`/api/db/v0/tables/${id}/split_table/`, {
+    extract_columns: idsOfColumnsToExtract,
+    extracted_table_name: extractedTableName,
+  });
+}
+
+export function moveColumns(
+  tableId: number,
+  idsOfColumnsToMove: number[],
+  targetTableId: number,
+): CancellablePromise<null> {
+  return postAPI(`/api/db/v0/tables/${tableId}/move_columns/`, {
+    move_columns: idsOfColumnsToMove,
+    target_table: targetTableId,
+  });
+}
+
 /**
  * Replace getTable with this function once the above mentioned changes are done.
  */
-export function getTableFromStore(
+export async function getTableFromStoreOrApi(
   id: TableEntry['id'],
-): CancellablePromise<TableEntry> {
+): Promise<TableEntry> {
```

Comment:
I prefer to return `CancellablePromise` in our store methods. There are many scenarios where CancellablePromises are beneficial, a few of them:
* A typeahead search: 2 requests are made to the same endpoint, request 2 resolves before request 1. The most recent updates expected by the user would be overwritten by the request 1 resolving. CancellablePromise can avoid that.
* Components that are destroyed containing promises that are not resolved, aren't removed from memory.
* A common component would need to cancel it's promises. Consider the TableView, when route changes, the component is not re-rendered, the same component clears it's states. It needs to cancel all previous requests inorder to not interfere with the new ones.

We should probably make this a standard.

Reply:
Ok that makes sense. It's a drag that composing `CancellablePromise` requires so much boilerplate compared to `Promise` where we can use `async`/`await`, but I can't think of a way around that. I've fixed this in c6e58001d86f88bb1f1fbe69d81b682215f19074